testing/preact.js

```diff
@@ -0,0 +1,63 @@
+import * as preact from /*OK*/ 'preact';
+
+/**
+ * This file introduces a helper for draining Preact's queue of renders and effects.
+ * We use this as part of the afterEach() cleanup in unit tests, to ensure no effects are run
+ * in subsequent tests.
+ *
+ * There is still a test isolation issue in that an effect can asynchronously schedule work
+ * which cannot be guarded from at this layer. For that we'd likely need to refresh the window
+ * in between each test run.
+ *
+ * We should be able to remove this file if this feature lands in Preact.
+ * @fileoverview
+ */
+
+const rafs = [];
+/**
+ * @param {(ts: (DOMHighResTimeStamp) => number)} cb
+ * @return {number}
+ */
+function flushableRaf(cb) {
+  rafs.push(cb);
+  return requestAnimationFrame(flushRaf);
+}
+
+function flushRaf(ts = performance.now()) {
+  while (rafs.length) {
+    rafs.shift()(ts);
+  }
+}
+
+let pendingRender;
+
+/**
+ * @param {() => void} process
+ * @return {Promise<void>}
+ */
+async function flushableRender(process) {
+  pendingRender = () => {
+    pendingRender = null;
+    return process();
+  };
+  await Promise.resolve().then(pendingRender);
+}
+
+/**
+ * Flushes Preact renders and effects.
+ *
+ * Effects may queue up further rerenders, etc. etc,
+ * so this function will loop until everything to resolves.
+ *
+ * @return {Promise<void>}
+ */
+export async function flush() {
+  await flushRaf();
+  while (pendingRender) {
+    await pendingRender();
+    await flushRaf();
```

Comment:
There's a time-order bug if we await here, but luckily we don't need to await a sync function.

If we awaited here, then a render queued by an effect will be processed before we flush in this loop. That's this await takes 2 ticks (even for a sync return value), and the `flushableRender`'s pendingRender will itself fire on the 2nd tick. If the second render queued an effect, then we won't notice to recall `flushRaf`.
```suggestion
   flushRaf();
```